lib/mzgl/util/Listenable.h

```diff
@@ -12,25 +12,20 @@ template <class T>
 class Listenable {
 public:
 	void addListener(T *listener) {
-		for (int i = 0; i < listeners.size(); i++) {
-			if (listeners[i] == listener) {
-				return;
-			}
+		if (std::find(listeners.begin(), listeners.end(), listener) != listeners.end()) {
+			return;
 		}
+
 		listeners.push_back(listener);
 	}
 
 	void removeListener(T *listener) {
-		for (int i = 0; i < listeners.size(); i++) {
-			if (listeners[i] == listener) {
-				listeners.erase(listeners.begin() + i);
-				return;
-			}
-		}
+		listeners.erase(std::remove(listeners.begin(), listeners.end(), listener), listeners.end());
 	}
+	[[nodiscard]] int getNumListeners() const { return listeners.size(); }
 
 protected:
```

Comment:
Could this be private?

Reply:
(It probably should be, or we need to add a virtual destructor)

Reply:
It needs to be protected because you subclass listenable then directly access the listeners from the subclass to fire the events.

Reply:
will add virtual desctructors as per discord convo